docs/src/_docs/samplers.jmd

```diff
@@ -0,0 +1,246 @@
+    bumps = sin(m) + cos(m)
+    m = m + 5*bumps
+    for i in eachindex(x)
+      x[i] ~ Normal(m, sqrt(s))
+    end
+    return s, m
+end
+
+# Define our data points.
+x = [1.5, 2.0, 13.0, 2.1, 0.0]
+
+# Set up the model call, sample from the prior.
+model = gdemo(x)
+vi = Turing.VarInfo()
+model(vi, Turing.SampleFromPrior())
+vi.flags["trans"] = [true, false]
+
+# Evaluate surface at coordinates.
+function evaluate(m1, m2)
+    vi.vals .= [m1, m2]
+    model(vi, Turing.SampleFromPrior())
+    -vi.logp
+end
+
+function plot_sampler(chain)
+    # Extract values from chain.
+    ss = link.(Ref(InverseGamma(2,3)), chain[:s])
+    ms = chain[:m]
+    lps = chain[:lp]
+
+    # How many surface points to sample.
+    granularity = 500
+
+    # Range start/stop points.
+    spread = 0.5
+    σ_start = minimum(ss) - spread * std(ss); σ_stop = maximum(ss) + spread * std(ss);
+    μ_start = minimum(ms) - spread * std(ms); μ_stop = maximum(ms) + spread * std(ms);
+    σ_rng = collect(range(σ_start, stop=σ_stop, length=granularity))
+    μ_rng = collect(range(μ_start, stop=μ_stop, length=granularity))
+
+    # Make surface plot.
+    p = surface(σ_rng, μ_rng, evaluate,
+          camera=(30, 65),
+          ticks=nothing,
+          colorbar=false,
+          color=:inferno)
+
+    line_range = 1:length(ms)
+
+    plot3d!(ss[line_range], ms[line_range], -lps[line_range],
+        lc =:viridis, line_z=collect(line_range),
+        legend=false, colorbar=false, alpha=0.5)
+
+    return p
+end
+```
+
+## Samplers
+
+### Gibbs
+
+Gibbs sampling tends to exhibit a "jittery" trajectory. The example below combines `HMC` and `PG` sampling to traverse the posterior.
+
+```julia
+c = sample(model, Gibbs(1000,
+  HMC(1, 0.01, 5, :s), PG(20, 1, :m)))
+plot_sampler(c)
+```
+
+Other samplers can be combined as well:
+
+```julia
+c = sample(model, Gibbs(1000, MH(1, :s), SGLD(100, 0.01, :m)))
+plot_sampler(c)
+```
+
+### HMC
+
+Hamiltonian Monte Carlo (HMC) sampling is a typical sampler to use, as it tends to be fairly good at converging in a efficient manner. It can often be tricky to set the correct parameters for this sampler however, and the `NUTS` sampler is often easier to run if you don't want to spend too much time fiddling with step size and and the number of steps to take.
+
+```julia
+c = sample(model, HMC(1000, 0.01, 10))
+plot_sampler(c)
+```
+
+### HMCDA
+
+The HMCDA sampler is an implementation of the Hamiltonian Monte Carlo with Dual Averaging algorithm found in the paper "The No-U-Turn Sampler: Adaptively Setting Path Lengths in Hamiltonian Monte Carlo" by Hoffman and Gelman (2011). The paper can be found on [arXiv](https://arxiv.org/abs/1111.4246) for the interested reader.
+
+```julia
+c = sample(model, HMCDA(1000, 200, 0.65, 0.3))
+plot_sampler(c)
+```
+
+### IS
+
+Importance Sampling (IS) attempts to identify parameters that appear more important to the posterior, and sample them more often.
+
+```julia
+c = sample(model, IS(1000))
+plot_sampler(c)
```

Comment:
I won't include IS here since all samples are weighted. So the scatter/path plot does not make sense.